### PR TITLE
Flux language style guide

### PR DESCRIPTION
**Pull request by scottrigby**
- See https://github.com/fluxcd/flux2/discussions/620

### CODE REVIEW

**Review by dholbach — APPROVED** [review]
This looks great to me - let's have one more look over it together later on in the Flux Dev meeting.

File: style-guide.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    - GitOps Toolkit
+    - Flagger
+        - part of the Flux family of projects. Can also be described as a Flux sub-project
+        - Can use text from the top of GitHub `fluxcd/flagger` README. Either just the first sentence or the first paragraph, and optionally any of the following sentences there as needed:
+            > Flagger is a progressive delivery tool that automates the release process for applications running on Kubernetes. It reduces the risk of introducing a new software version in production by gradually shifting traffic to the new version while measuring metrics and running conformance tests.
+            >
+            > Flagger implements several deployment strategies (Canary releases, A/B testing, Blue/Green mirroring) using a service mesh (App Mesh, Istio, Linkerd, Open Service Mesh) or an ingress controller (Contour, Gloo, NGINX, Skipper, Traefik) for traffic routing. For release analysis, Flagger can query Prometheus, Datadog, New Relic or CloudWatch and for alerting it uses Slack, MS Teams, Discord, Rocket and Google Chat.
+            >
+            > Flagger is a Cloud Native Computing Foundation project and part of Flux family of GitOps tools.
```
Comment (squaremo) [inline style-guide.md:32]:
```suggestion
            > Flagger is a Cloud Native Computing Foundation project and part of the Flux family of GitOps tools.
```

File: style-guide.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
+# Flux style guide
+
+This style guide defines how and in which contexts different Flux descriptions may be used.
+
+## Disambiguity
```
Comment (squaremo) [inline style-guide.md:5]:
```suggestion
## Disambiguation
```

Comment (scottrigby) [reply style-guide.md:5]: Ha yes thanks!

File: style-guide.md [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+There is some understandable ambiguity when describing Flux in different contexts. For example, "Flux" is the name of a CNCF project which includes Flagger as a sub-project. There is a `flux` CLI command. Flux is sometimes described as being comprised of controller components, which (in addition to reusable Go packages?) we call the GitOps Toolkit. Flux is also sometimes used to refer only to the CLI + controllers, without the optional Flagger sub-project.
+
+These should be clarified, and defined consistently within our own docs and readmes, if we care how Flux is understood and described by other projects, media, etc.
+
+1. "Flux" CNCF incubating project
```
[the remark this answers is not in the document]
Comment (scottrigby) [reply style-guide.md:11]: @squaremo It was just meant to show that the word "Flux" is used in multiple ways, and must be disambiguated. Though maybe it's redundant to continue those quotes throughout the ordered list. Will remove them.

[the remark this answers is not in the document]
Comment (scottrigby) [reply style-guide.md]: Done

**Review by dholbach — APPROVED** [review]
Thanks for this PR @scottrigby

**Comment by dholbach** [comment]
Can you please rebase and squash?

**Comment by pjbgf** [comment]
Relates to https://github.com/fluxcd/website/issues/587

**Pull request merged**